protocol/v1/qbft/instance/instance.go

```diff
@@ -257,44 +258,24 @@ func (i *Instance) ProcessMsg(msg *specqbft.SignedMessage) (bool, error) {
 		return false, errors.Wrap(err, "invalid signed message")
 	}
 
-	// TODO(nkryuchkov): remove error wrapping, make the processing similar,
-	// optionally remove pipelines and use the code from spec
-	switch msg.Message.MsgType {
-	case specqbft.ProposalMsgType:
-		if err := i.ProposalMsgPipeline().Run(msg); err != nil {
-			if errors.Is(err, signedmsg.ErrWrongRound) {
-				// NOTE: These 4 checks will be replaced by one in a future PR.
-				i.Logger.Debug(fmt.Sprintf("message round (%d) does not equal state round (%d)", msg.Message.Round, i.State().GetRound()))
-			}
-			return false, fmt.Errorf("invalid proposal message: %w", err)
-		}
-	case specqbft.PrepareMsgType:
-		if err := i.PrepareMsgPipeline().Run(msg); err != nil {
-			if errors.Is(err, signedmsg.ErrWrongRound) {
-				// NOTE: These 4 checks will be replaced by one in a future PR.
-				i.Logger.Debug(fmt.Sprintf("message round (%d) does not equal state round (%d)", msg.Message.Round, i.State().GetRound()))
-			}
-			return false, err
-		}
-	case specqbft.CommitMsgType:
-		if err := i.CommitMsgPipeline().Run(msg); err != nil {
-			if errors.Is(err, signedmsg.ErrWrongRound) {
-				// NOTE: These 4 checks will be replaced by one in a future PR.
-				i.Logger.Debug(fmt.Sprintf("message round (%d) does not equal state round (%d)", msg.Message.Round, i.State().GetRound()))
-			}
-			return false, err
-		}
-	case specqbft.RoundChangeMsgType:
-		if err := i.ChangeRoundMsgPipeline().Run(msg); err != nil {
-			if errors.Is(err, signedmsg.ErrWrongRound) {
-				// NOTE: These 4 checks will be replaced by one in a future PR.
-				i.Logger.Debug(fmt.Sprintf("message round (%d) does not equal state round (%d)", msg.Message.Round, i.State().GetRound()))
-			}
-			return false, fmt.Errorf("invalid round change message: %w", err)
-		}
-	default:
+	pipelineMap := map[specqbft.MessageType]pipelines.SignedMessagePipeline{
+		specqbft.ProposalMsgType:    i.ProposalMsgPipeline(),
```

Comment:
`ProcessMsg` func should be light. init a map and hold all pipelines i think is over kill. 
why no keep switch case and just handle the errors at the end instead of in each case?

Reply:
@nivBlox 
The map looked to me easier to read than a switch. But a switch is fine too. However, I'd prefer to handle all errors in the end to avoid having 4 same checks like this:
```go
if errors.Is(err, signedmsg.ErrWrongRound) {
	// NOTE: These 4 checks will be replaced by one in a future PR.
	i.Logger.Debug(fmt.Sprintf("message round (%d) does not equal state round (%d)", msg.Message.Round, i.State().GetRound()))
}
```

Reply:
@nivBlox yes thats what i meant

Reply:
@nivBlox ok, I'll change to a switch now

Reply:
@nivBlox done